src/_posts/hands-on-iii.md

```diff
@@ -0,0 +1,334 @@
+
+A solution would have been to declare those values as part of the schema but it does not work out of the box. It requires a call to the entity’s [`flushToDOM()`](https://aframe.io/docs/0.5.0/core/entity.html#flushtodom-recursive) method because A-Frame avoids serializing data for performance reasons. Furthermore, it is currently impossible to properly serialize the parent node if that node doesn’t have an `id`.
+
+One way of preserving this data, without changing the way the element is used, is to store it inside the HTML node. The important piece is in the `init` method where the component creates or reuses a storage object to hold the state.
+
+```js
+AFRAME.registerComponent('tool', {
+  schema: {
+    position: { type: 'vec3' },
+    rotation: { type: 'vec3' }
+  },
+  init() {
+    this.el._tool = this.el._tool || {};
+  },
+  grab() {
+    this._rememberLocation();
+    this.el.setAttribute('position', this.data.position);
+    this.el.setAttribute('rotation', this.data.rotation);
+    this.el._tool._isGrabbed = true;
+  },
+  drop() {
+    this._restoreLocation();
+    this.el._tool._isGrabbed = false;
+  },
+  _rememberLocation() {
+    this.el._tool._originalParent = this.el.parentNode;
+    this.el._tool._originalPosition = this.el.getAttribute('position');
+    this.el._tool._originalRotation = this.el.getAttribute('rotation');
+  },
+  _restoreLocation() {
+    this.el._tool._originalParent.appendChild(this.el);
+    this.el.setAttribute('position', this.el._tool._originalPosition);
+    this.el.setAttribute('rotation', this.el._tool._originalRotation);
+  },
+  isGrabbed() {
+    return this.el._tool._isGrabbed;
+  }
+});
+```
+
+Now we can grab every element in the scene:
+
+[![Grabbing tools](/images/blog/hands-on/step3-grabbing-tools-thumb.png)](/images/blog/hands-on/step3-grabbing-tools.gif)
+
+### Fixing the collision bounding shapes
+
+In the [previous article](/blog/hands-on-ii.html), I mentioned that the physic engine was miscalculating the collision bounding box of the VIVE controllers. Making the bounding box to match the controller model is especially important when trying to grab physical objects, after disabling the debug mode of physics. If the bounding box, and the model position and dimensions don’t match, the user could not predict the distances for which grabbing something is possible resulting in a counterintuitive and frustrating interaction.
```

Comment:
`the user could not predict the distances for which grabbing something is possible resulting in a counterintuitive and frustrating interaction.` maybe `the user interactions won't match what the user sees resulting in a frustrating experience`